File: Library/core/src/main/java/com/emedinaa/core/media/PicassoImageLoader.java
package com.emedinaa.core.media;

import android.widget.ImageView;

import com.emedinaa.core.media.transform.CircleTransform;
import com.squareup.picasso.Picasso;

import java.io.File;

/**
 * @author : Eduardo Medina
 * @see : https://github.com/emedinaa/android-without-libraries/tree/dev/MeetupApp/MeetupApp/app/src/main/java/com/emedinaa/meetupapp/common/media
 * @since : 8/11/18
 */
public class PicassoImageLoader implements ImageLoader{

    @Override
    public void load(String url, ImageView imageView) {
        Picasso.get()
                .load(url)
                .into(imageView);
    }

    @Override
    public void loadCircle(String url, ImageView imageView) {
        Picasso.get()
                .load(url)
                .transform(new CircleTransform())
                .into(imageView);
    }

    @Override
    public void loadLocal(String path, ImageView imageView) {
        Picasso.get()
                .load(new File(path))
                .into(imageView);
    }
}

File: RestaurantApp/app/src/main/java/com/emedinaa/kotlinapp/util/PicassoImageLoader.kt
package com.emedinaa.kotlinapp.util

import android.widget.ImageView
import com.squareup.picasso.Picasso
import java.io.File

class PicassoImageLoader:ImageLoader {

    override fun load(url: String, imageView: ImageView) {
        Picasso.get()
            .load(url)
            .into(imageView)
    }

    override fun loadWithResize(url: String, imageView: ImageView, resize: Int) {
        Picasso.get()
            .load(url)
            .resize(resize,resize)
            .into(imageView)
    }

    override fun loadCircle(url: String, imageView: ImageView) {
        Picasso.get()
            .load(url)
            .transform(CircleTransform())
            .into(imageView)
    }

    override fun loadLocal(path: String, imageView: ImageView) {
        Picasso.get()
            .load(File(path))
            .into(imageView)
    }
}